sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.integration.service.ts

```diff
@@ -178,6 +250,37 @@ export class IER12IntegrationService extends SFTPIntegrationBase<void> {
     return null;
   }
 
+  /**
+   * Get scholastic standing code.
+   * Scholastic standing code does not consider the
+   * scholastic standing change type `Student withdrew from program`.
+   * @param assessmentTriggerType
+   * @param scholasticStandingChangeType
+   * @returns scholastic standing code.
+   */
+  private getScholasticStandingCode(
+    assessmentTriggerType: AssessmentTriggerType,
+    scholasticStandingChangeType: StudentScholasticStandingChangeType,
+  ): ScholasticStandingCode | null {
+    if (
+      assessmentTriggerType !==
+        AssessmentTriggerType.ScholasticStandingChange ||
+      scholasticStandingChangeType ===
+        StudentScholasticStandingChangeType.StudentWithdrewFromProgram
+    ) {
+      return null;
+    }
+    const scholasticStandingCodeMap = {
+      [StudentScholasticStandingChangeType.StudentDidNotCompleteProgram]:
+        ScholasticStandingCode.UC,
+      [StudentScholasticStandingChangeType.StudentCompletedProgramEarly]:
+        ScholasticStandingCode.EC,
+      [StudentScholasticStandingChangeType.ChangeInIntensity]:
+        ScholasticStandingCode.CI,
+    };
+    return scholasticStandingCodeMap[scholasticStandingChangeType];
```

Comment:
I would prefer a `switch` with a default in case there is a missing type. I believe that once the code passes line 271 a value is expected to be returned, right?
```ts
switch (scholasticStandingChangeType) {
  case StudentScholasticStandingChangeType.StudentDidNotCompleteProgram:
    return ScholasticStandingCode.UC;
  case StudentScholasticStandingChangeType.StudentCompletedProgramEarly:
    return ScholasticStandingCode.EC;
  case StudentScholasticStandingChangeType.ChangeInIntensity:
    return ScholasticStandingCode.CI;
  default:
    throw new Error(
      `Invalid 'Scholastic Standing Change Type' ${scholasticStandingChangeType}`,
    );
}
```

Reply:
Deepak and I talked and since it was just a recommendation the code will stay as it is.
@guru-aot FYI since you added the "thumbs up".